src/OrchardCore.Modules/OrchardCore.ContentTypes/Deployment/ContentDefinitionDeploymentSource.cs

```diff
@@ -37,8 +43,8 @@ public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlan
             result.Steps.Add(new JsonObject
             {
                 ["name"] = "ContentDefinition",
-                ["ContentTypes"] = JArray.FromObject(contentTypes),
-                ["ContentParts"] = JArray.FromObject(contentParts),
+                ["ContentTypes"] = JArray.FromObject(contentTypes, _jsonSerializerOptions),
```

Comment:
@sebastienros do you think this also isn't needed?
```
    public class ContentTypeDefinitionRecord
    {
        public ContentTypeDefinitionRecord()
        {
            ContentTypePartDefinitionRecords = [];
        }

        public string Name { get; set; }

        public string DisplayName { get; set; }

        public JsonObject Settings { get; set; }

        public IList<ContentTypePartDefinitionRecord> ContentTypePartDefinitionRecords { get; set; }
    }
```

Reply:
Not, ContentTypeDefinition doesn't have anything polymorphic. I think we should handle it in a case by case basis and only use it when we know it's necessary. And keep the comments which state that it's used be we know there are or can be "dynamic" polymorphic types in the json document.

For instance if the types were well-known and would not come from a different assembly, we could even list them on the base type without having to register them in the options.

Reply:
And see this `JsonObject Settings`. It's not even considered polymoprhic, it will just contain whatever is in the json without loss. Then the thing that uses the setting will deserialize this node with custom options if it needs it.